pkgs/development/idris-modules/idris-wrapper.nix

```diff
@@ -1,14 +1,12 @@
-{ symlinkJoin, makeWrapper, stdenv }: idris: { path, lib }:
+{ lib, symlinkJoin, makeWrapper, idris-no-deps, gcc, gmp }:
 
 symlinkJoin {
-  name = idris.name;
-  src = idris.src;
-  paths = [ idris ];
+  inherit (idris-no-deps) name src meta;
+  paths = [ idris-no-deps ];
   buildInputs = [ makeWrapper ];
-  meta.platforms = idris.meta.platforms;
   postBuild = ''
     wrapProgram $out/bin/idris \
-      --suffix PATH : ${ stdenv.lib.makeBinPath path } \
-      --suffix LIBRARY_PATH : ${stdenv.lib.makeLibraryPath lib}
-      '';
-  }
+      --run 'export IDRIS_CC=''${IDRIS_CC:-${lib.getBin gcc}/bin/gcc}' \
```

Comment:
Can we use `makeWrapper`'s `--set` parameter here? See:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh#L16

Reply:
Unfortunately not, `--set` really sets it equal to exactly the string you give it, can't expand any variables (it uses bash's `${parameter@operator}` "Parameter transformation" with operator=Q, see `man bash`). `--run` is the next best way.

Reply:
I see. All good then.